### PR TITLE
API / CoW: constructing Series from Series creates lazy copy (with default copy=False)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/pandas-dev/pandas/issues/48998 (bullet point about constructors)

In the CoW proposal, the idea for constructors would be that those follow the lazy copy with CoW approach as well, just like other methods and indexing operations that create a new Series/DataFrame from another Series/DataFrame.  
That was not yet implemented in the initial implementation, and this PR is adding it for the `Series(series)` case.

Currently, the way I implemented this is by making a shallow copy of the manager. This however also affects the default (non-CoW) behaviour -> see https://github.com/pandas-dev/pandas/issues/49523. 
Depending on whether we are OK with changing ("fixing"?) that behaviour in 2.0 or not, I can leave this as is, or otherwise put the copy behind a `if get_cnofig("mode.copy_on_write")` check.

- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by phofl — COMMENTED** [review]
lgtm, I would be ok with changing the behavior

**Review by mroeschke — COMMENTED** [review]
LGTM too. Could use a whatsnew note

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -77,6 +77,13 @@ be set to ``"pyarrow"`` to return pyarrow-backed, nullable :class:`ArrowDtype` (
         df_pyarrow = pd.read_csv(data, use_nullable_dtypes=True, engine="pyarrow")
     df_pyarrow.dtypes
 
+Copy on write improvements
+^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v2.0.0.rst:81]: Added a new section like is being added in https://github.com/pandas-dev/pandas/pull/50471. We can consolidate those different blurbs later on in a coherent whatsnew.

**Comment by jorisvandenbossche** [comment]
I updated this to put the actual code change behind a "if CoW" check (so this PR only changes something for the case of CoW enabled), and will do a separate PR to address https://github.com/pandas-dev/pandas/issues/49523 (and for both Series and DataFrame).

**Comment by jorisvandenbossche** [comment]
> will do a separate PR to address #49523 (and for both Series and DataFrame).

Opened https://github.com/pandas-dev/pandas/pull/50539 for this

**Pull request merged**